### PR TITLE
Better defaults when reached end of classic blocks

### PR DESCRIPTION
**Pull request by joshuacolvin0**
* Force snapshot save to database when final classic block is reached
* Default to no pruning on arbitrum one and Rinkeby testnet since non-archive classic node isn't useful
* Cleanup some log messages

### CODE REVIEW

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 1]
```diff
@@ -131,6 +106,37 @@ bool ArbCore::checkError() {
     return core_error;
 }
 
+void ArbCore::printMachineOutputInfo(const std::string& msg,
+                                     MachineOutput& machine_output) {
+    std::string l2_block_string;
+    if (machine_output.l2_block_number > coreConfig.final_block) {
+        l2_block_string = "TERMINAL_BLOCK";
```
Comment (PlasmaPower) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:113]: Won't this always print TERMINAL_BLOCK if final_block is the default of zero?

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 2]
```diff
@@ -1717,7 +1743,8 @@ bool ArbCore::threadBody(ThreadDataStruct& thread_data) {
                     std::chrono::seconds(coreConfig.database_save_interval);
             }
 
-            if (trigger_save_rocksdb_checkpoint ||
+            if (output.l2_block_number >= coreConfig.final_block ||
```
Comment (PlasmaPower) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:1746]: Same here with the default being zero

**Review by PlasmaPower — COMMENTED** [review]
Generally LGTM but I think the final_block default of zero causes some issues

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp (hunk 1 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp:114]: Fixed

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp (hunk 2 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp]: Fixed

**Review by PlasmaPower — APPROVED** [review]
LGTM

**Pull request merged**